src/tooltip/Tooltip.tsx

```diff
@@ -1,357 +1,223 @@
-    };
-  };
-
-  getTooltipStyle = () => {
-    const { yOffset, xOffset, elementHeight, elementWidth } = this.state;
-    const {
-      height,
-      backgroundColor,
-      width,
-      withPointer,
-      containerStyle,
-    } = this.props;
-    const { x, y } = getTooltipCoordinate(
-      xOffset,
-      yOffset,
-      elementWidth,
-      elementHeight,
-      ScreenWidth,
-      ScreenHeight,
-      width,
-      height,
-      withPointer
-    );
-
-    return StyleSheet.flatten([
-      {
-        position: 'absolute',
-        [I18nManager.isRTL ? 'right' : 'left']: x,
-        top: y,
-        width,
-        height,
-        backgroundColor,
-        // default styles
-        display: 'flex',
-        alignItems: 'center',
-        justifyContent: 'center',
-        flex: 1,
-        borderRadius: 10,
-        padding: 10,
-      },
-      containerStyle,
-    ]);
-  };
-
-  renderPointer = (tooltipY: FlexStyle['top']) => {
-    const { yOffset, xOffset, elementHeight, elementWidth } = this.state;
-    const { backgroundColor, pointerColor } = this.props;
-    const pastMiddleLine = yOffset > (tooltipY || 0);
-
-    return (
-      <View
-        style={{
-          position: 'absolute',
-          top: pastMiddleLine ? yOffset - 13 : yOffset + elementHeight - 2,
-          [I18nManager.isRTL ? 'right' : 'left']:
-            xOffset +
-            getElementVisibleWidth(elementWidth, xOffset, ScreenWidth) / 2 -
-            7.5,
-        }}
-      >
-        <Triangle
-          style={{ borderBottomColor: pointerColor || backgroundColor }}
-          isDown={pastMiddleLine}
-        />
-      </View>
-    );
-  };
-
-  getTooltipHighlightedButtonStyle = (): ViewStyle => {
-    const { highlightColor } = this.props;
-    const { yOffset, xOffset, elementWidth, elementHeight } = this.state;
-
-    return {
-      position: 'absolute',
-      top: yOffset,
-      [I18nManager.isRTL ? 'right' : 'left']: xOffset,
-      backgroundColor: highlightColor,
-      overflow: 'visible',
-      width: elementWidth,
-      height: elementHeight,
-    };
-  };
-
-  renderTouchableHighlightedButton = () => {
-    const TooltipHighlightedButtonStyle = this.getTooltipHighlightedButtonStyle();
-
-    return (
-      <TouchableOpacity
-        testID="tooltipTouchableHighlightedButton"
-        onPress={() => this.toggleTooltip()}
-        style={TooltipHighlightedButtonStyle}
-      >
-        {this.props.children}
-      </TouchableOpacity>
-    );
-  };
-
-  renderStaticHighlightedButton = () => {
-    const TooltipHighlightedButtonStyle = this.getTooltipHighlightedButtonStyle();
-
-    return (
-      <View style={TooltipHighlightedButtonStyle}>{this.props.children}</View>
-    );
-  };
-
-  renderHighlightedButton = () => {
-    const { closeOnlyOnBackdropPress } = this.props;
-    if (closeOnlyOnBackdropPress) {
-      return this.renderTouchableHighlightedButton();
+const ToolTip2: React.FunctionComponent<ToolTip2Props> = ({
+  children,
```

Comment:
Can we add support for showing tooltip only on hover. 

Like this

<img width="510" alt="Screenshot 2021-06-12 at 5 25 03 PM" src="https://user-images.githubusercontent.com/32242596/121775091-21ac9b80-cba3-11eb-943d-bd7f718a18e2.png">

Reply:
Can be done on web only

Reply:
Can we use pressable state onPressIn and hide it on press out

Reply:
I will try